typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts

```diff
@@ -25,29 +27,36 @@ export class EV5InterchainAccountTxTransformer
 
   public async transform(
     ...txs: PopulatedTransaction[]
-  ): Promise<PopulatedTransaction[]> {
+  ): Promise<ethers.PopulatedTransaction[]> {
     const txChainsToInnerCalls: Record<ChainName, CallData[]> = {};
 
     txs.map(({ to, data, value, chainId }: PopulatedTransaction) => {
-      assert(to, 'Invalid PopulatedTransaction: Missing to field');
-      assert(data, 'Invalid PopulatedTransaction: Missing data field');
-      assert(chainId, 'Invalid PopulatedTransaction: Missing chainId field');
       const txChain = this.multiProvider.getChainName(chainId);
       if (!txChainsToInnerCalls[txChain]) txChainsToInnerCalls[txChain] = [];
       txChainsToInnerCalls[txChain].push({ to, data, value });
     });
```

Comment:
instead of pushing to the array and accounting for empty in here, you can just have the map return the new array

Reply:
oh nvm I see this is how you are doing the group by destinationChain

Reply:
one alternative might be map then flatmap or reduce or something but this is more of a taste thing
I like eliminating unnecessary branches where possible

Reply:
I'm not sure on how to remove the branch entirely, but this provided great inspo for both using `reduce`, which eliminated line 29, and cleaning up the branch, as such:
```
const txChainsToInnerCalls: Record<ChainName, CallData[]> = txs.reduce((
    txChainToInnerCalls: Record<ChainName, CallData[]>,
    { to, data, value, chainId }: PopulatedTransaction,
) => {
    const txChain = this.multiProvider.getChainName(chainId);
    txChainToInnerCalls[txChain] ||= [];
    txChainToInnerCalls[txChain].push({ to, data, value });
    return txChainToInnerCalls;
}, {});
```

Reply:
I think its generally an antipattern in a map/reduce function that you are declaring with the return value

Reply:
the inner var `txChainToInnerCalls` (singular) and the assigned return var `txChainsToInnerCalls` (plural) are different

is this what you're referring to?